Src/Support/Google.Apis.Auth.AspNetCore/GoogleCredentialProvider.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Google.Apis.Auth.OAuth2;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Auth.AspNetCore
+{
+    public class GoogleCredentialProvider : IGoogleCredentialProvider
+    {
+        public GoogleCredentialProvider(IServiceProvider services, string authenticationScheme)
+        {
+            _services = services;
+            _authenticationScheme = authenticationScheme;
+        }
+
+        private readonly IServiceProvider _services;
+        private readonly string _authenticationScheme;
+
+        public async Task<GoogleCredential> GetAsync(CancellationToken cancellationToken)
+        {
+            // TODO: Should failures throw an exception? Or return null (as currently coded)?
+            var httpContext = _services.GetRequiredService<IHttpContextAccessor>().HttpContext;
+            var auth = await httpContext.AuthenticateAsync(_authenticationScheme);
+            if (!auth.Succeeded || auth.None)
+            {
+                // Auth failed, cannot provide a credential.
+                // TODO: Logging
+                return null;
+            }
+            var optionsMonitor = _services.GetRequiredService<IOptionsMonitor<OpenIdConnectOptions>>();
+            var clock = _services.GetRequiredService<ISystemClock>();
+            var authProperties = auth.Properties;
+            var options = optionsMonitor.Get(OpenIdConnectDefaults.AuthenticationScheme);
```

Comment:
Here is used  
```csharp
OpenIdConnectDefaults.AuthenticationScheme
```

 while in builder used by default

```csharp
GoogleOpenIdConnectDefaults.AuthenticationScheme
``` 

If they both refer to the same identity service, shouldn't be Google here by default?

Reply:
Thanks for spotting this; you may well be correct. I'll see if I can continue working on this soon.